### ISSUE TITLE
Computer/Pocket Computer file system protection

### ISSUE THREAD

**Issue by m-doescode** [SquidDev-CC/CC-Tweaked#2043]
Excuse me if this has already been mentioned before, I tried but failed to find any issue relating to this.

Adding a setting to block mounting the computer's files through a disk drive. As you know, you can put a pocket computer or regular computer into a disk drive and any attached computer can then access this computer's files. This is fine, but it means that you can't really protect your computer from being infiltrated.

The idea is that a setting could be added, maybe fs.allow_drive_mounting. When added to a disk drive, the mod checks to see if this setting is set to false in the .settings file, and if so, refuses to mount the computer. This would effectively prevent any access to the computer unless you're the owner and know the password/entry sequence.

On the other hand, it gives a false sense of security. Your computer can always be griefed, not inspected, sure, but it can be broken and that's still just as bad. Also, it makes it seem like it's completely inaccessible but of course any server admin with server file access can read your files, so it may confuse players.

Finally, it breaks the principle that if you own the computer, you get to decide what it runs. What if you forgot to turn off disk startup and someone installs a virus on it? Now you can't recover it at all without an admin's help. Logically, if you have physical access, you should at least be able to wipe the computer to not have your gold go to waste (although it isn't really a lot of gold...)

All that said, it can still be useful if you trust your admins and have a mod similar to securitycraft that protects computers from being broken.

What do you think?

**Comment by Lupus590** [comment]
You could just encrypt your important files.
Also, don't put your pocket computer into untrusted disk drives.
Likewise, don't put untrusted disks into your disk drives.

**Comment by m-doescode** [comment]
> You could just encrypt your important files.

Ehhh not really much of a solution given how inefficient encryption is, and it's quite advanced for a simple player that just wants a a way to secure their computer.

> Also, don't put your pocket computer into untrusted disk drives.

Sure, but what's stopping someone from mining your computer and putting it into one?

**Comment by zyxkad** [comment]
> Sure, but what's stopping someone from mining your computer and putting it into one?

claims, or server rules

**Comment by zyxkad** [comment]
> Logically, if you have physical access, you should at least be able to wipe the computer to not have your gold go to waste (although it isn't really a lot of gold...)

Yes, that's why you can access computer files in a disk drive without any permissions.

**Comment by SquidDev** [comment]
Woops, pressed enter too early, sorry!

Thank for the suggestion! I understand the motivation, but I'm afraid this is not something I want to add.

> Finally, it breaks the principle that if you own the computer, you get to decide what it runs. What if you forgot to turn off disk startup and someone installs a virus on it? 

I think this is the crux of it really. I think if you mess up your computer (either deliberately or accidentally), there needs to be a way to recover from it (well, short of just deleting the files). The ability to mount a computer in the drive is incredibly useful, and something I think that needs to always be available.

**Comment by m-doescode** [comment]
Seems like a satisfactory conclusion has been reached, so I'll close the issue
